static/scripts/assignment_table.js

```diff
@@ -0,0 +1,482 @@
+                indicators: true,
+            },
+            hiddenRows: {
+                rows: [0],
+                indicators: true,
+            },
+            contextMenu: ['commentsAddEdit', 'commentsRemove', 'hidden_columns_hide', 'hidden_rows_hide', 'hidden_columns_show', 'hidden_rows_show'],
+            filters: true,
+            dropdownMenu: ['filter_by_value', 'filter_action_bar', 'undo'],
+            className: 'controlsQuickFilter htCenter htMiddle',
+            colHeaders: allHeaders,
+            columns: columns,
+            colWidths: 100,
+            columnHeaderHeight: 225,
+            rowHeaders: fixedRowsText,
+            rowHeaderWidth: 125,
+            init: function () {
+                if (comments) {
+                    comments.forEach(comment => {
+                        this.setCellMeta(comment.row, comment.col, comment.key, comment.value);
+                    });
+                }
+                isCollectedMetaData = false;
+            },
+            afterGetColHeader: function (col, th) {
+                th.style.transform = 'rotate(180deg)';
+                th.style.writingMode = 'vertical-lr';
+                th.style.textAlign = 'center';
+                th.style.fontWeight = 'bold';
+                th.style.height = '100px';
+                th.style.lineHeight = '100px';
+
+                if (col > lenFixedHeaders - 1) {
+                    let colData = this.getDataAtCol(col);
+                    if (colData[RowIndices.TOTAL_ASSISTANT_NOW] >= colData[RowIndices.ASSISTANTS]) {
+                        th.style.backgroundColor = 'green';
+                    }
+                }
+            },
+            afterChange: function (changes) {
+                if (changes) {
+                    changes.forEach(([row, prop, oldValue, newValue]) => {
+                        let col = this.propToCol(prop);
+                        if ((col >= lenFixedHeaders && row >= lenFixedRowsText) && (row % 2 === 1)) {
+
+                            const colInfos = this.getDataAtProp(prop);
+                            let nbrAssistants = colInfos[RowIndices.TOTAL_ASSISTANT_NOW];
+                            const rowInfos = this.getDataAtRow(row - 1);
+                            const quadri = colInfos[RowIndices.QUADRI];
+
+                            let loadKey = quadri === 1 ? ColumnIndices.LOAD_Q1 : ColumnIndices.LOAD_Q2;
+                            let loadValue = rowInfos[loadKey];
+
+                            // Old non-empty value and new empty value: loadValue-- and nbrAssistants--.
+                            // Old non-empty value and new non-empty value: loadValue++ and nbrAssistants++.
+                            // Old null value and new non-empty value: loadValue++ and nbrAssistants++.
+                            // Old non-null value and new null value: loadValue-- and nbrAssistants--.
+                            if (oldValue !== newValue) {
```

Comment:
We still need a small doc for this in the UI. It won't be clear for user that whatever the cell content is corresponds to *assigned* and that an empty cell corresponds to *not assigned*.

Another nice and valid solution would be to catch double click for instance and *check* or *uncheck* the cell.